src/timing.c

```diff
@@ -325,20 +350,27 @@ void jl_timing_init_task(jl_task_t *t)
 #endif
 }
 
+int cmp_name_idx(const void *name, const void *idx) {
+    return strcmp((const char *)name, jl_timing_names[*(const int *)idx]);
+}
+
 JL_DLLEXPORT int jl_timing_set_enable(const char *subsystem, uint8_t enabled)
 {
-    for (int i = 0; i < JL_TIMING_LAST; i++) {
-        if (strcmp(subsystem, jl_timing_names[i]) == 0) {
-            uint64_t subsystem_bit = (1ul << i);
-            if (enabled) {
-                jl_timing_enable_mask |= subsystem_bit;
-            } else {
-                jl_timing_enable_mask &= ~subsystem_bit;
-            }
-            return 0;
-        }
+    const int *idx = (const int *)bsearch(subsystem, jl_timing_names_sorted, JL_TIMING_EVENT_LAST, sizeof(int), cmp_name_idx);
+    if (idx == NULL)
+        return -1;
+    int i = *idx;
+    // sorted names include events, so skip if we're looking at an event instead of a subsystem
+    // events are always at least JL_TIMING_LAST
+    if (i >= JL_TIMING_LAST)
```

Comment:
Hmm, what guarantees that the subsystems are the first `JL_TIMING_LAST` items after sorting?

Reply:
This is guaranteed by the JL_TIMING_EVENTS macro, which puts subsystems before events.

Reply:
Isn't that affected by the sort?

Reply:
Oh I just saw that `jl_timing_names_sorted` array is an array of indices, not a sorted version of the original array

Understanding that, this looks correct to me.